File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/python/exercises/practice/transpose/transpose.py
def transpose(text):
    lines = text.split('\n')
    if not lines:
        return ''
    max_length = max(len(line) for line in lines)
    transposed = []
    for i in range(max_length):
        row = []
        for line in lines:
            if i < len(line):
                row.append(line[i])
            else:
                row.append(' ')
        transposed_row = ''.join(row)
        transposed.append(transposed_row)
    return '\n'.join(transposed)

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/java/exercises/practice/transpose/src/main/java/Transpose.java
public class Transpose {
    public String transpose(String toTranspose) {
        if (toTranspose.isEmpty()) {
            return "";
        }
        
        String[] lines = toTranspose.split("\n");
        int maxLen = 0;
        for (String line : lines) {
            if (line.length() > maxLen) {
                maxLen = line.length();
            }
        }
        
        StringBuilder result = new StringBuilder();
        for (int col = 0; col < maxLen; col++) {
            StringBuilder transposedRow = new StringBuilder();
            int lastNonPaddingIndex = -1;
            for (int i = 0; i < lines.length; i++) {
                String line = lines[i];
                if (col < line.length()) {
                    char c = line.charAt(col);
                    transposedRow.append(c);
                    // Track last original character (even spaces) that's not padding
                    lastNonPaddingIndex = transposedRow.length() - 1;
                } else {
                    transposedRow.append(' ');
                }
            }
            // Trim only trailing padding spaces
            String row = lastNonPaddingIndex >= 0 
                ? transposedRow.substring(0, lastNonPaddingIndex + 1)
                : "";
            if (result.length() > 0) {
                result.append('\n');
            }
            result.append(row);
        }
        
        return result.toString();
    }
}
